### PR TITLE
Add downloadURL config option

### PR DESCRIPTION
**Pull request by arudge**
Accepts a valid java.net.URI for mongodb download

### CODE REVIEW

**Comment by arudge** [comment]
Addressed the comments from https://github.com/sourcemuse/GradleMongoPlugin/pull/27

File: build.gradle [hunk 1]
```diff
@@ -6,12 +6,14 @@ apply plugin: 'signing'
...
 archivesBaseName = 'gradle-mongo-plugin'
 group = 'com.sourcemuse.gradle.plugin'
-version = '0.9.0-SNAPSHOT'
+version = '0.9.1-SNAPSHOT'
 
 sourceCompatibility = 1.7
 targetCompatibility = 1.7
 
 repositories {
+    mavenLocal()
```
Comment (roberttaylor426) [inline build.gradle]: Not keen to add a dependency to mavenLocal(), we want builds to be repeatable. Would you mind dropping this?

File: build.gradle [hunk 2]
```diff
@@ -6,12 +6,14 @@ apply plugin: 'signing'
 
 archivesBaseName = 'gradle-mongo-plugin'
 group = 'com.sourcemuse.gradle.plugin'
-version = '0.9.0-SNAPSHOT'
+version = '0.9.1-SNAPSHOT'
```
Comment (roberttaylor426) [inline build.gradle]: 0.9.0 hasn't been released yet, so would you mind leaving the version as-is?

(I'm happy to cut a release as soon as this PR is merged).

File: src/test/groovy/com/sourcemuse/gradle/plugin/GradleMongoPluginExtensionSpec.groovy [hunk 3]
```diff
@@ -123,4 +123,25 @@ class GradleMongoPluginExtensionSpec extends Specification {
         throwable.getMessage() == "MongodVerbosity should be defined as either '-verbose' or '-v(vvvv)'. " +
                 "Do not configure this property if you don't wish to have verbose output."
     }
+
+    def 'mongod download url throws exception for invalid url'() {
+        given:
+        String invalidURL = 'thisisnotavalidurl'
```
Comment (roberttaylor426) [inline src/test/groovy/com/sourcemuse/gradle/plugin/GradleMongoPluginExtensionSpec.groovy]: Tiny comment, but would you mind keeping the spacing between the `given` and `when` blocks? It just keeps things consistent with the other tests and helps readability.

**Comment by roberttaylor426** [comment]
A couple of small comments, otherwise LGTM.

File: build.gradle [hunk 4]
```diff
@@ -6,12 +6,14 @@ apply plugin: 'signing'
...
 
 sourceCompatibility = 1.7
 targetCompatibility = 1.7
 
 repositories {
+    mavenLocal()
+    jcenter()
```
Comment (roberttaylor426) [inline build.gradle]: Why do we need this?

**Comment by arudge** [comment]
Thanks for the review!
I addressed all your concerns, but let me know if I missed one.

Adding `jcenter()` allows dependency resolution from 'https://bintray.com/'. 
Nothing critical, so I removed it. 

If you want to create some Github issues for any new features or house keeping items, I could contribute some additional pull requests.

File: src/test/groovy/com/sourcemuse/gradle/plugin/GradleMongoPluginExtensionSpec.groovy [hunk 5]
```diff
@@ -123,4 +123,27 @@ class GradleMongoPluginExtensionSpec extends Specification {
...
+        this.pluginExtension.downloadURL = invalidURL
+
+        then:
+        def throwable = thrown(IllegalArgumentException)
+        throwable.message == "DownloadURL ${invalidURL} is not a valid URI."
+    }
+
+    def 'mongod download url can be set for valid url'() {
+        given:
+        String invalidURL = 'http://google.com'
```
Comment (roberttaylor426) [inline src/test/groovy/com/sourcemuse/gradle/plugin/GradleMongoPluginExtensionSpec.groovy]: `validUrl` surely? :-)

**Comment by roberttaylor426** [comment]
Thanks Andy that would be great. I was planning to address #28 by the end of this week, but if you want to beat me to it please feel free. :-)

One more small comment added btw.

File: src/test/groovy/com/sourcemuse/gradle/plugin/GradleMongoPluginExtensionSpec.groovy (hunk 5 above)
Comment (arudge) [inline src/test/groovy/com/sourcemuse/gradle/plugin/GradleMongoPluginExtensionSpec.groovy]: bah, I really shouldn't try to get out PRs while kids are running around.

**Comment by arudge** [comment]
Ok, should be good now

**Pull request merged**